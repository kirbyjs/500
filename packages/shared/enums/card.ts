export enum CardTypes {
    JOKER = 'JOKER',
    ACE = 'ACE',
    KING = 'KING',
    QUEEN = 'QUEEN',
    JACK = 'JACK',
    TEN = 'TEN',
    NINE = 'NINE',
    EIGHT = 'EIGHT',
    SEVEN = 'SEVEN',
    SIX = 'SIX',
    FIVE = 'FIVE',
    FOUR = 'FOUR'
}

export enum CardSuits {
    HEART = 'HEART',
    DIAMOND = 'DIAMOND',
    CLUB = 'CLUB',
    SPADE = 'SPADE'
}
