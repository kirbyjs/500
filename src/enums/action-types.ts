export enum RequestActionTypes {
    CREATE_SESSION = 'CREATE_SESSION',
    JOIN_SESSION = 'JOIN_SESSION',
    BID = 'BID',
    PLAY = 'PLAY'
}

export enum ResponseActionTypes {
    ADD_USER = 'ADD_USER',
    REMOVE_USER = 'REMOVE_USER',
    UPDATE_CARDS_IN_TRICK = 'UPDATE_CARDS_IN_TRICK',
    UPDATE_PLAY_SCORE = 'UPDATE_PLAY_SCORE',
    UPDATE_GAME_SCORE = 'UPDATE_GAME_SCORE'
}
