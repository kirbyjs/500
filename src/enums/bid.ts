export enum BidTypes {
    SPADES = 'SPADES',
    CLUBS = 'CLUBS',
    DIAMONDS = 'DIAMONDS',
    HEARTS = 'HEARTS',
    NO_TRUMP = 'NO_TRUMP',
    MISERE = 'MISERE'
}

export enum BidNumbers {
    SIX = 6,
    SEVEN = 7,
    EIGHT = 8,
    NINE = 9,
    TEN = 10
}
