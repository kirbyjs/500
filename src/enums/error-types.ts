enum ActionLambdaErrorTypes {
    EMPTY_REQUEST_BODY = 'EMPTY_REQUEST_BODY'
}

export default ActionLambdaErrorTypes;
