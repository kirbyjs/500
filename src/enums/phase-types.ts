enum PhaseTypes {
    SESSION = 'SESSION',
    BID = 'BID',
    PLAY = 'PLAY'
}

export default PhaseTypes;
