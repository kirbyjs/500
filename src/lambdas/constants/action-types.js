// Created by kirbyjs on 2/9/20.

export const CREATE_SESSION = 'CREATE_SESSION';
export const JOIN_SESSION = 'JOIN_SESSION';
export const BID = 'BID';
export const PLAY = 'PLAY';
