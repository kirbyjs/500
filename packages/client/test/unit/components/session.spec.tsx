import { TextField } from '@material-ui/core';
import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { random, name } from 'faker';
import App from 'src/app';
import Session from 'src/components/session';
import useStores from 'src/hooks/use-stores';
import { RequestActionTypes } from 'shared/enums/action-types';
import gameFactory from '../factories/game';
import viewStateFactory from '../factories/view-state';
import userFactory from '../factories/user';

jest.mock('src/hooks/use-stores');
const useStoresMock = useStores as jest.Mock;

describe('<Session />', () => {
    let component: ShallowWrapper<typeof App>;

    beforeEach(() => {
        const game = gameFactory.build();
        const viewState = viewStateFactory.build();

        useStoresMock.mockReturnValue({ game, viewState });
        component = shallow(<Session />);
    });

    it('should render Session', () => {
        expect(component).toMatchSnapshot();
    });

    it('should render form elements', () => {
        expect(component.childAt(0).dive()).toMatchSnapshot();
    });

    it('should use the hook useStores and pass required session state', () => {
        const game = gameFactory.build({
            contextualUser: userFactory.build({
                name: name.firstName()
            })
        });
        const viewState = viewStateFactory.build({
            selectedSessionTabIndex: random.boolean() ? 0 : 1
        });

        useStoresMock.mockReturnValue({ game, viewState });

        // when
        component = shallow(<Session />);
        const reactFragmentComponent = component.childAt(0).dive();
        const tabsComponent = reactFragmentComponent.childAt(0);
        const userTextField = reactFragmentComponent.childAt(1).childAt(0);

        // then
        expect(tabsComponent.prop('value')).toBe(viewState.selectedSessionTabIndex);
        expect(tabsComponent.prop('onChange')).toBe(viewState.setSelectedSessionTabIndex);

        expect(userTextField.prop('value')).toBe(game.contextualUser.name);
        expect(userTextField.prop('onChange')).toBe(game.contextualUser.setUsername);
    });

    it('should render the game id text input field if join session tab is selected', () => {
        const game = gameFactory.build({
            id: random.uuid()
        });
        const viewState = viewStateFactory.build({
            selectedSessionTab: RequestActionTypes.JOIN_SESSION
        });

        useStoresMock.mockReturnValue({ game, viewState });

        // when
        component = shallow(<Session />);
        const reactFragmentComponent = component.childAt(0).dive();
        const gameIdTextField = reactFragmentComponent.childAt(1).childAt(1);

        // then
        expect(gameIdTextField.type()).toBe(TextField);
        expect(gameIdTextField.props()).toStrictEqual({
            id: 'gameId',
            label: 'Game Id',
            onChange: game.setGameId,
            value: game.id
        });
    });
});
