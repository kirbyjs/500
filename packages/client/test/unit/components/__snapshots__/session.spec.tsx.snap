// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Session /> should render Session 1`] = `
<form
  id="session"
>
  <Observer>
    <Component />
  </Observer>
  <WithStyles(ForwardRef(Button))
    color="primary"
    id="create-session-button"
    size="large"
    startIcon={
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              640,
              512,
              Array [],
              "f11b",
              "M480.07 96H160a160 160 0 1 0 114.24 272h91.52A160 160 0 1 0 480.07 96zM248 268a12 12 0 0 1-12 12h-52v52a12 12 0 0 1-12 12h-24a12 12 0 0 1-12-12v-52H84a12 12 0 0 1-12-12v-24a12 12 0 0 1 12-12h52v-52a12 12 0 0 1 12-12h24a12 12 0 0 1 12 12v52h52a12 12 0 0 1 12 12zm216 76a40 40 0 1 1 40-40 40 40 0 0 1-40 40zm64-96a40 40 0 1 1 40-40 40 40 0 0 1-40 40z",
            ],
            "iconName": "gamepad",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
    }
    type="button"
    variant="contained"
  >
    Press Start
  </WithStyles(ForwardRef(Button))>
</form>
`;

exports[`<Session /> should render form elements 1`] = `
<Fragment>
  <WithStyles(ForwardRef(Tabs))
    indicatorColor="primary"
    onChange={[Function]}
    textColor="primary"
    value={0}
  >
    <WithStyles(ForwardRef(Tab))
      label="Create"
    />
    <WithStyles(ForwardRef(Tab))
      label="Join"
    />
  </WithStyles(ForwardRef(Tabs))>
  <div
    className="form-inputs"
  >
    <WithStyles(ForwardRef(TextField))
      id="username"
      label="Name"
      onChange={[Function]}
      value="Balzee"
    />
  </div>
</Fragment>
`;
